cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_v2.rs

```diff
@@ -315,156 +264,215 @@ fn transfer_relay_token() {
+
+		let assets = vec![weth_asset, dot_asset.clone(), local_fee_asset.clone()];
+
+		let xcms = VersionedXcm::from(Xcm(vec![
+			WithdrawAsset(assets.clone().into()),
+			PayFees { asset: local_fee_asset.clone() },
+			InitiateTransfer {
+				destination: destination(),
+				remote_fees: Some(AssetTransferFilter::ReserveWithdraw(Definite(
+					remote_fee_asset.clone().into(),
+				))),
+				preserve_origin: true,
+				assets: vec![
+					AssetTransferFilter::ReserveWithdraw(Definite(reserve_asset.clone().into())),
+					AssetTransferFilter::ReserveDeposit(Definite(dot_asset.into())),
+				],
+				remote_xcm: Xcm(vec![DepositAsset {
+					assets: Wild(All),
+					beneficiary: beneficiary(),
+				}]),
+			},
+		]));
+
+		<AssetHubWestend as AssetHubWestendPallet>::PolkadotXcm::execute(
+			RuntimeOrigin::signed(AssetHubWestendReceiver::get()),
+			bx!(xcms),
+			Weight::from(EXECUTION_WEIGHT),
+		)
+		.unwrap();
 	});
 
 	BridgeHubWestend::execute_with(|| {
 		type RuntimeEvent = <BridgeHubWestend as Chain>::RuntimeEvent;
-		type RuntimeOrigin = <BridgeHubWestend as Chain>::RuntimeOrigin;
-
-		// Register WND on BH
-		assert_ok!(<BridgeHubWestend as BridgeHubWestendPallet>::Balances::force_set_balance(
-			RuntimeOrigin::root(),
-			MultiAddress::Id(BridgeHubWestendSender::get()),
-			INITIAL_FUND * 10,
-		));
-		assert_ok!(<BridgeHubWestend as BridgeHubWestendPallet>::EthereumSystem::register_token(
-			RuntimeOrigin::root(),
-			Box::new(VersionedLocation::from(Location::parent())),
-			AssetMetadata {
-				name: "wnd".as_bytes().to_vec().try_into().unwrap(),
-				symbol: "wnd".as_bytes().to_vec().try_into().unwrap(),
-				decimals: 12,
-			},
-		));
+		// Check that Ethereum message was queue in the Outbound Queue
 		assert_expected_events!(
 			BridgeHubWestend,
-			vec![RuntimeEvent::EthereumSystem(snowbridge_pallet_system::Event::RegisterToken { .. }) => {},]
+			vec![RuntimeEvent::EthereumOutboundQueueV2(snowbridge_pallet_outbound_queue_v2::Event::MessageQueued{ .. }) => {},]
 		);
+	});
+}
+
+#[test]
+fn create_agent() {
+	fund_sovereign();
+
+	register_weth();
 
-		// Transfer some WETH to AH
-		let message = VersionedMessage::V1(MessageV1 {
-			chain_id: CHAIN_ID,
-			command: Command::SendToken {
-				token: WETH.into(),
-				destination: Destination::AccountId32 { id: AssetHubWestendReceiver::get().into() },
-				amount: TOKEN_AMOUNT,
-				fee: XCM_FEE,
+	BridgeHubWestend::execute_with(|| {});
+
+	AssetHubWestend::execute_with(|| {
+		type RuntimeOrigin = <AssetHubWestend as Chain>::RuntimeOrigin;
+
+		let local_fee_asset =
+			Asset { id: AssetId(Location::parent()), fun: Fungible(LOCAL_FEE_AMOUNT_IN_DOT) };
+
+		// All WETH as fee and reserve_asset is zero, so there is no transfer in this case
+		let remote_fee_asset = Asset { id: AssetId(weth_location()), fun: Fungible(TOKEN_AMOUNT) };
+		let reserve_asset = Asset { id: AssetId(weth_location()), fun: Fungible(0) };
```

Comment:
reserve_asset as zero, so there is no transfer and all burnt WETH on AH are used as fee in this case.